test/e2e/metrics_k8s_cluster_receiver_test.go

```diff
@@ -0,0 +1,154 @@
+				Build()
+			objs = append(objs, &pipelineOnlyContainerMetricsEnabled)
+
+			backendPodMetricsEnabled := backend.New(mockNs, backend.SignalTypeMetrics, backend.WithName(backendK8sClusterPodMetricsEnabledName))
+			objs = append(objs, backendPodMetricsEnabled.K8sObjects()...)
+			backendK8sClusterPodMetricsEnabledURL = backendPodMetricsEnabled.ExportURL(proxyClient)
+
+			pipelineOnlyPodMetricsEnabled := testutils.NewMetricPipelineBuilder().
+				WithName(pipelineK8sClusterPodMetricsEnabledName).
+				WithRuntimeInput(true).
+				WithRuntimeInputContainerMetrics(false).
+				WithOTLPOutput(testutils.OTLPEndpoint(backendPodMetricsEnabled.Endpoint())).
+				Build()
+			objs = append(objs, &pipelineOnlyPodMetricsEnabled)
+
+			return objs
+		}
+
+		BeforeAll(func() {
+			k8sObjects := makeResources()
+
+			DeferCleanup(func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+			})
+			Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		It("Should have healthy pipelines", func() {
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineK8sClusterContainerMetricsEnabledName)
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineK8sClusterPodMetricsEnabledName)
+		})
+
+		It("Should have metrics backends running", func() {
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendK8sClusterContainerMetricsEnabledName, Namespace: mockNs})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendK8sClusterPodMetricsEnabledName, Namespace: mockNs})
+		})
+
+		It("Should deliver runtime container metrics to container-metrics backend", func() {
+			Eventually(func(g Gomega) {
+				resp, err := proxyClient.Get(backendK8sClusterContainerMetricsEnabledURL)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				bodyContent, err := io.ReadAll(resp.Body)
+				defer resp.Body.Close()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				containerMetricNames := slices.Concat(kubeletstats.ContainerMetricsNames, k8scluster.ContainerMetricsNames)
+				g.Expect(bodyContent).To(WithFlatMetrics(WithNames(
+					OnlyContainElementsOf(containerMetricNames))), "Found container metrics in backend that are not part of k8scluster or kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(k8scluster.ContainerMetricsNames)))), "Found container metrics in backend that are not part of k8scluster")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(kubeletstats.ContainerMetricsNames)))), "Found container metrics in backend that are not part of kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(WithScopeAndVersion(ContainElement(And(
+					HaveField("Name", metric.InstrumentationScopeK8sCluster),
+					HaveField("Version",
+						SatisfyAny(
+							ContainSubstring("main"),
```

Comment:
Can you change the checks for the version to be similar to what we have [here](https://github.com/kyma-project/telemetry-manager/blob/3c3d15ff2eec2da4072ebaa5e6be51723981f73a/test/e2e/metrics_kyma_input_test.go#L111-L112). I know that we have tests using the same checks as the one you used, but they will break once we have a new major version (2.x.x) and we don't have versions starting with `PR-`. I'm working on a separate PR to align the checks for the scope version in all tests.

Reply:
also, same thing in the version check in the test below `Should deliver runtime pod metrics to pod-metrics backend`